### PR TITLE
Update cicd docs

### PR DESCRIPTION
**Pull request by kolodziejczak**
**Description**

Changes proposed in this pull request:

- Update cicd docs

### CODE REVIEW

File: docs/contributor/04-30-ci-cd.md [hunk 1]
```diff
@@ -1,51 +1,70 @@
...
-| [`pre-istio-presubmit-scanner`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/test-infra/presubmit-scanner.yaml#L470)                                  | Runs the Gitleaks presubmit scanner to detect any sensitive data that might have been committed.                                                                              |
-| [`Pull Request integration tests / Kubernetes version compatibility test`](https://github.com/kyma-project/istio/blob/main/.github/workflows/pull-integration.yaml#L33) | Executes the basic integration test suite for the Istio module and checks if any incompatibilities occurred during the execution.                       |
+| [`pull-istio-operator-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/kyma-project/istio/istio-manager.yaml#L51)                                 | Builds Kyma Istio Operator's image and pushes it to the `dev` registry.                                                                                |
```
[the remark this answers is not in the document]
Comment (kolodziejczak) [reply docs/contributor/04-30-ci-cd.md:11]: right

File: docs/contributor/04-30-ci-cd.md [hunk 2]
```diff
@@ -1,51 +1,71 @@
 # CI/CD
 The Istio module contains several pipelines that run on pull requests, on the main branch, on the release branch, and periodically. They perform various tasks such as running a linter, executing unit tests with code coverage, generating artifacts for development and production purposes, and running integration tests. The integration tests ensure the functional correctness of Kyma Istio Operator and verify if upgrading to the new version of Kyma Istio Operator is possible.
 
-## Pipelines running on pull requests
+## Pipelines running on development pull requests
```
Comment (triffer) [inline docs/contributor/04-30-ci-cd.md:4]: We should ahve it similar to [api-gateway](https://github.com/kyma-project/api-gateway/blob/main/docs/contributor/04-30-ci-cd.md#pipelines-running-on-pull-requests-to-the-main-branch)

```suggestion
## Pipelines running on pull requests to the `main` branch
```

File: docs/contributor/04-30-ci-cd.md [hunk 3]
```diff
@@ -1,51 +1,71 @@
...
+| [`main integration tests / Wait for image build`](https://github.com/kyma-project/istio/blob/26d8cfa69b91828fc34afd3b6aaf63c854d0cc54/.github/workflows/main-integration.yaml#L20)            | Waits for the operator's image build to finish so that image-dependent jobs can be executed.                                                      |
+| [`main integration tests / Istio upgrade integration tests`](https://github.com/kyma-project/istio/blob/26d8cfa69b91828fc34afd3b6aaf63c854d0cc54/.github/workflows/main-integration.yaml#L30) | Executes the upgrade integration test suite of Istio Module and verifies if the existing release can be successfully upgraded to the new version. |
+| [`main integration tests / Slack Notification`](https://github.com/kyma-project/istio/blob/26d8cfa69b91828fc34afd3b6aaf63c854d0cc54/.github/workflows/main-integration.yaml#L44)              | Sends a Slack notification to the team's channel if any pipelines fail on the main branch.                                                        |
+
+## Pipelines running on the PR during the release process
```
Comment (triffer) [inline docs/contributor/04-30-ci-cd.md:36]: We should have it similar to [api-gateway](https://github.com/kyma-project/api-gateway/blob/main/docs/contributor/04-30-ci-cd.md#pipelines-running-on-pull-requests-to-release-branches).
```suggestion
## Pipelines running on pull requests to release branches
```

**Review by triffer — COMMENTED** [review]
I saw that we are not documenting anything for the [release workflow](https://github.com/kyma-project/istio/blob/main/.github/workflows/create-release.yaml). But I think the reason is, that's a manual trigger to execute an action and not used to verify the correctness.

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**